main/config.c

```diff
@@ -2767,18 +2807,56 @@ int ast_config_text_file_save2(const char *configfile, const struct ast_config *
 	/* go thru all the inclusions and make sure all the files involved (configfile plus all its inclusions)
 	   are all truncated to zero bytes and have that nice header*/
 	for (incl = cfg->includes; incl; incl = incl->next) {
-		if (!incl->exec) { /* leave the execs alone -- we'll write out the #exec directives, but won't zero out the include files or exec files*/
-			/* normally, fn is just set to incl->included_file, prepended with config dir if relative */
-			fi = set_fn(fn, sizeof(fn), incl->included_file, configfile, fileset);
-			f = fopen(fn, "w");
-			if (f) {
-				gen_header(f, configfile, fn, generator);
-				fclose(f); /* this should zero out the file */
+		if (incl->include_type <= 0) {
+			/* leave the #exec's alone -- we'll write out the #exec directives, but won't zero out the exec files */
+			make_fn(fn, sizeof(fn), incl->included_file, configfile);
+			if (incl->include_type == 0) {
+				/* A #include. */
+
+				/* normally, fn is just set to incl->included_file, prepended with config dir if relative */
+				fi = set_fn(fn, sizeof(fn), incl->included_file, configfile, fileset);
+				f = fopen(fn, "w");
+				if (f) {
+					ast_verb(2, "Saving include '%s'\n", fn);
+					gen_header(f, configfile, fn, generator);
+					fclose(f); /* this should zero out the file */
+				} else {
+					ast_log(LOG_ERROR, "Unable to write %s (%s)\n", fn, strerror(errno));
+				}
+				if (fi) {
+					ao2_ref(fi, -1);
+				}
 			} else {
-				ast_log(LOG_ERROR, "Unable to write %s (%s)\n", fn, strerror(errno));
-			}
-			if (fi) {
-				ao2_ref(fi, -1);
+				int glob_ret;
+				glob_t globbuf;
+				
+				/* A #tryinclude. */
+				globbuf.gl_offs = 0;
+				glob_ret = glob(fn, MY_GLOB_FLAGS, NULL, &globbuf);
+				if (glob_ret == 0) {
+					int i;
+					
+					for (i = 0; i < globbuf.gl_pathc; i++) {
+						if (access(globbuf.gl_pathv[i], F_OK) != 0) {
+							/* skip this #tryinclude path as we don't not want
+							   to create files that do not already exist. */
+							continue;
+						}
+						fi = set_fn(fn, sizeof(fn), globbuf.gl_pathv[i], configfile, fileset);
+						f = fopen(fn, "w");
+						if (f) {
+							ast_verb(2, "Saving tryinclude '%s'\n", fn);
```

Comment:
same with this one

Reply:
I was following the existing code that writes out "configfile" (pre-changes: line 2795).